src/overlays/actors/ovl_Dm_Ravine/z_dm_ravine.c

```diff
@@ -19,16 +25,57 @@ const ActorInit Dm_Ravine_InitVars = {
     (ActorFunc)DmRavine_Init,
     (ActorFunc)DmRavine_Destroy,
     (ActorFunc)DmRavine_Update,
-    (ActorFunc)DmRavine_Draw
+    (ActorFunc)DmRavine_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Ravine_0x80A2E7A0/DmRavine_Init.asm")
+void DmRavine_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DmRavine* this = THIS;
+    u8 flag = gSaveContext.perm.weekEventReg[0];
+    if (((flag & 0x10) | cREG(0)) != 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    this->loaded = false;
+    globalCtx->roomContext.unk7A[0] = 1;
+    globalCtx->roomContext.unk7A[1] = 0;
+    this->state = 0;
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actionFunc = &DmRavine_Action;
+}
+
+void DmRavine_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Ravine_0x80A2E7A0/DmRavine_Destroy.asm")
+// This function is unused, but takes two word-sized arguments
+// `thisx` & `globalCtx` are guesses; but it could be something else
+void DmRavine_Action(Actor* thisx, GlobalContext* globalCtx) {
```

Comment:
My other comment on this seems to be gone, but I agree with this being `DmRavine_DoNothing(DmRavine* this, GlobalContext* globalCtx)`